src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -40,8 +40,11 @@ public override NoPia.EmbeddedTypesManager EmbeddedTypesManagerOpt
         /// </summary>
         private Dictionary<FieldSymbol, NamedTypeSymbol> _fixedImplementationTypes;
 
-        private bool _needsGeneratedIsReadOnlyAttribute_IsFrozen;
         private bool _needsGeneratedIsReadOnlyAttribute_Value;
+        private bool _needsGeneratedIsByRefLikeAttribute_Value;
+
+        private bool _needsGeneratedIsReadOnlyAttribute_IsFrozen;
+        private bool _needsGeneratedIsByRefLikeAttribute_IsFrozen;
```

Comment:
I have tried that by using ThreeState value here. However it appears we permit changing from false to true and back unless frozen is set, but frozen could match either true or false. 

I am not quite sure that is actually the expected behavior, but decided to not change that in this change to not break things unintentionally. 
i will consider replacing the "frozen bool" with a more generic implementation in a separate change. 

logged [https://github.com/dotnet/roslyn/issues/19809](https://github.com/dotnet/roslyn/issues/19809) to track this.


---
In reply to: [118734363](https://github.com/dotnet/roslyn/pull/19795#discussion_r118734363) [](ancestors = 118734363)